src/main/java/org/sagebionetworks/bridge/hibernate/HibernateSurveyDao.java

```diff
@@ -0,0 +1,365 @@
+        System.out.println("12");
+        return createSurvey(copy);
+    }
+
+    @Override
+    public Survey publishSurvey(StudyIdentifier studyIdentifier, Survey survey, boolean newSchemaRev) {        
+        if (!survey.isPublished()) {
+            // update survey
+            survey.setPublished(true);
+            survey.setModifiedOn(DateUtils.getCurrentMillisFromEpoch());
+            // make schema from survey
+            if (!survey.getUnmodifiableQuestionList().isEmpty()) {
+                UploadSchema schema = uploadSchemaService.createUploadSchemaFromSurvey(studyIdentifier, survey, newSchemaRev);
+                survey.setSchemaRevision(schema.getRevision());
+            }
+            updateSurvey(studyIdentifier, survey);
+        }
+        return survey;
+    }
+
+    @Override
+    public void deleteSurvey(Survey survey) {
+        survey.setDeleted(true);
+        hibernateHelper.update(survey, null);
+    }
+    
+    @Override
+    public void deleteSurveyPermanently(StudyIdentifier studyIdentifier, GuidCreatedOnVersionHolder keys) {
+        Survey existing = getSurvey(studyIdentifier, keys, false);
+        
+        if (existing != null) {
+            deleteAllElements(existing.getGuid(), existing.getCreatedOn());
+            
+            SurveyId surveyId = new SurveyId(keys);
+            hibernateHelper.deleteById(HibernateSurvey.class, surveyId);
+            
+            // Delete the schemas as well, or they accumulate.
+            try {
+                StudyIdentifier studyId = new StudyIdentifierImpl(existing.getStudyIdentifier());
+                uploadSchemaService.deleteUploadSchemaByIdPermanently(studyId, existing.getIdentifier());
+            } catch(EntityNotFoundException e) {
+                // This is OK. Just means this survey wasn't published.
+            }
+        }
+    }
+    
+    @Override
+    public Survey getSurvey(StudyIdentifier studyIdentifier, GuidCreatedOnVersionHolder keys, boolean includeElements) {
+        if (keys.getGuid().toLowerCase().startsWith(IDENTIFIER_PREFIX)) {
+            ImmutableMap<String,Object> params = ImmutableMap.of(
+                    "studyKey", studyIdentifier.getIdentifier(),
+                    "identifier", keys.getGuid().substring(IDENTIFIER_PREFIX.length()),
+                    "createdOn", keys.getCreatedOn());
+            
+            String getQuery = SELECT_TEMPLATE + GET_ACTIVE_WITH_IDENTIFIER;
+            
+            List<HibernateSurvey> results = hibernateHelper.queryGet(
+                    getQuery, params, OFFSET, PAGESIZE, HibernateSurvey.class);
+            
+            if (!results.isEmpty()) {
+                return results.get(0);
+            }
+            return null;
+        }
+        SurveyId surveyId = new SurveyId(keys);
+        Survey ret = hibernateHelper.getById(HibernateSurvey.class, surveyId);
+        if (ret != null && includeElements) {
+            attachSurveyElements(ret);
+        }
```

Comment:
The reason we had the includeElements flag in DynamoSurveyDao is that querying the second table was very expensive. In SQL, it's just another query, and Hibernate handles it for us, so we no longer need that flag. Also, since we already fetched survey elements, it's probably unnecessary to query for survey elements a second time.

That said, the survey is probably just the raw survey elements, and it would be useful to have a helper method to convert those into the subclasses and to reconcile the survey rules.